modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/resource/releases/SW360Release.java

```diff
@@ -201,30 +210,77 @@ public void setCopyrights(String copyrights) {
+                        }
+                        coordinates = Collections.unmodifiableMap(coordinates);
+                    }
+            }
+        });
+        return this;
     }
 
     public SW360Release mergeWith(SW360Release releaseWithPrecedence) {
-        cpeid = Optional.of(releaseWithPrecedence.getCpeid()).orElse(cpeid);
-        downloadurl = Optional.of(releaseWithPrecedence.getDownloadurl()).orElse(downloadurl);
-        mainLicenseIds = Optional.of(releaseWithPrecedence.getMainLicenseIds()).orElse(mainLicenseIds);
-        coordinates = Optional.of(releaseWithPrecedence.getCoordinates()).orElse(coordinates);
-        finalLicense = Optional.of(releaseWithPrecedence.getFinalLicense()).orElse(finalLicense);
-        declaredLicense = Optional.of(releaseWithPrecedence.getDeclaredLicense()).orElse(declaredLicense);
-        observedLicense = Optional.of(releaseWithPrecedence.getObservedLicense()).orElse(observedLicense);
-        releaseTagUrl = Optional.of(releaseWithPrecedence.getReleaseTagUrl()).orElse(releaseTagUrl);
-        softwareHeritageId = Optional.of(releaseWithPrecedence.getSoftwareHeritageId()).orElse(softwareHeritageId);
-        hashes = Optional.of(releaseWithPrecedence.getHashes()).orElse(hashes);
-        clearingState = Optional.of(releaseWithPrecedence.getClearingState()).orElse(clearingState);
-        changeStatus = Optional.of(releaseWithPrecedence.getChangeStatus()).orElse(changeStatus);
-        copyrights = Optional.of(releaseWithPrecedence.getCopyrights()).orElse(copyrights);
-        externalIds = Optional.of(releaseWithPrecedence.getExternalIds()).orElse(externalIds);
+        cpeid = fun(releaseWithPrecedence, SW360Release::getCpeid);
+        downloadurl = fun(releaseWithPrecedence, SW360Release::getDownloadurl);
+        mainLicenseIds = fun(releaseWithPrecedence, SW360Release::getMainLicenseIds);
+        coordinates = fun(releaseWithPrecedence, SW360Release::getCoordinates);
+        finalLicense = fun(releaseWithPrecedence, SW360Release::getFinalLicense);
+        declaredLicense = fun(releaseWithPrecedence, SW360Release::getDeclaredLicense);
+        observedLicense = fun(releaseWithPrecedence, SW360Release::getObservedLicense);
+        releaseTagUrl = fun(releaseWithPrecedence, SW360Release::getReleaseTagUrl);
+        softwareHeritageId = fun(releaseWithPrecedence, SW360Release::getSoftwareHeritageId);
+        hashes = fun(releaseWithPrecedence, SW360Release::getHashes);
+        clearingState = fun(releaseWithPrecedence, SW360Release::getClearingState);
+        changeStatus = fun(releaseWithPrecedence, SW360Release::getChangeStatus);
+        copyrights =  fun(releaseWithPrecedence, SW360Release::getCopyrights);
 
         return this;
     }
 
+    private <T> T fun(SW360Release release, Function<SW360Release, T> getter) {
+        return Optional.ofNullable(getter.apply(release)).orElse(Optional.ofNullable(getter.apply(this)).orElse(null));
```

Comment:
better formatting
```suggestion
        return Optional.ofNullable(getter.apply(release))
            .orElse(Optional.ofNullable(getter.apply(this))
            .orElse(null));
```

And should still be equal to:
```suggestion
        return Optional.ofNullable(getter.apply(release))
            .orElse(getter.apply(this));
```

Reply:
As I said: Returns Nullpointerexception when `(getter.apply(this));` is null